xmps/color_selection/line_fits.py

```diff
@@ -0,0 +1,99 @@
+from astropy.io import fits
+import matplotlib
+matplotlib.use('Qt4Agg')
+import numpy as np
+import matplotlib.pyplot as plt
+
+from scipy.optimize import curve_fit
+from scipy import asarray as ar,exp
+import numpy.ma as ma
+
+'''
+This code is for fitting emission lines (gaussian) and measuring their equivalent widths
+'''
+
+Hb=4862.721
+oiii_4959=4960.295
+oiii_5007=5008.239
+Ha = 6564.614
+nii_6584=6585.27
+Hg=4341.68
+Hd=4102.89
+
+
+def gaus(x,a,x0,sigma):
+    return a*exp(-(x-x0)**2/(2*sigma**2))
+
+
+def LF(f,el):
+    # give a file and emission line
+    hdu=fits.open(f)
+    try:
+        wave=10**hdu[1].data['loglam']
+    except KeyError:
+        wave=hdu[1].data['LAMBDA'][0]
+    try:
+        flux=hdu[1].data['flux']
+    except KeyError:
+        flux=hdu[1].data['SPEC'][0]
+
+    error=(1/np.sqrt(ma.array(hdu[1].data['ivar'])))
+    if el=='oiii_4363':
+        el=4364.44
+    if el=='Hb':
+        el=4862.721
+    if el == 'oiii_4959':
+        el= 4960.295
+    if el == 'oiii_5007':
+        el=5008.239
+    if el ==  'Ha':
+        el= 6564.614
+    if el == 'nii_6584':
+        el = 6585.27
+    if el == 'Hg':
+        el =4341.68
+    if el == 'Hd':
+        el=4102.89
+
+    z=hdu[2].data['Z']
+    l=abs(wave-(z+1)*el)
+    pos=np.argmin(l)
+    pos2=np.argmin(abs(wave-(z+1)*6300))
+
+    peak_flux=flux[pos-8:pos+8]#-np.median(flux)
```

Comment:
Why +/- 8?   Should this be a variable?

Reply:
+/-8 seemed to be a good enough estimate for my spectra since line broadening wasn't much of a factor. ALIS had the final say with this as well. Perhaps we can discuss this further if need be.

Reply:
Ok, I think +/- 8  is fine.